src/commands/CreateDataSourceEntryCommand.ts

```diff
@@ -21,40 +21,60 @@ import * as vscode from 'vscode';
-		if (!entryType) {
+		let entryValue:string = await vscode.window.showInputBox( { placeHolder: 'Enter the value of the new property' });
+		if (!entryValue) {
 			return;
 		}
 
-		let refName: string;
-		let refKey: string;
-		if (entryType === constants.REFERENCE_TYPE_CONFIGMAP || entryType === constants.REFERENCE_TYPE_SECRET) {
-			refName = await vscode.window.showInputBox( { placeHolder: 'Enter the name of the reference' });
-			refKey = await vscode.window.showInputBox( { placeHolder: 'Enter the key for the reference' });
+		const yaml: DataVirtConfig = dsNode.getProject().dvConfig;
+		const file: string = dsNode.getProject().file;
+		let success: boolean = await handleDataSourceEntryCreation(yaml, dsNode.dataSourceConfig, file, constants.REFERENCE_TYPE_VALUE, entryName, entryValue, undefined, undefined);
+		if (success) {
+			vscode.window.showInformationMessage(`New datasource property ${entryName} has been created successfully...`);
 		} else {
-			refName = '';
-			refKey = '';
+			vscode.window.showErrorMessage(`An error occured when trying to create a new datasource property ${entryName} in datasource ${dsNode.label}...`);
+		}
+	}
+}
+
+export async function createDataSourceEntryCommandForSecret(dsNode: DataSourceTreeNode) {
+	await createDataSourceEntryCommandForReference(dsNode, constants.REFERENCE_TYPE_SECRET);
+}
+
+export async function createDataSourceEntryCommandForConfigMap(dsNode: DataSourceTreeNode) {
+	await createDataSourceEntryCommandForReference(dsNode, constants.REFERENCE_TYPE_CONFIGMAP);
+}
+
+async function createDataSourceEntryCommandForReference(dsNode: DataSourceTreeNode, type: string) {
+	if (dsNode) {
+		let entryName: string = await queryPropertyName(dsNode);
+		if (!entryName) {
+			return;
 		}
+
+		let refName: string = await vscode.window.showInputBox( { placeHolder: 'Enter the name of the reference' });
+		let refKey: string = await vscode.window.showInputBox( { placeHolder: 'Enter the key for the reference' });
+
 		// explicit check for undefined because in this case the user canceled the reference name or key input box
 		if (refName === undefined || refKey === undefined) {
 			return;
 		}
 
-		let entryValue:string = await vscode.window.showInputBox( { placeHolder: 'Enter the value of the new property' });
+		let entryValue:string = await vscode.window.showInputBox( { placeHolder: 'Enter the value of the new variable' });
 		if (!entryValue) {
 			return;
 		}
 
 		const yaml: DataVirtConfig = dsNode.getProject().dvConfig;
 		const file: string = dsNode.getProject().file;
-		let success: boolean = await handleDataSourceEntryCreation(yaml, dsNode.dataSourceConfig, file, entryType, entryName, entryValue, refName, refKey);
+		let success: boolean = await handleDataSourceEntryCreation(yaml, dsNode.dataSourceConfig, file, type, entryName, entryValue, refName, refKey);
 		if (success) {
 			vscode.window.showInformationMessage(`New datasource property ${entryName} has been created successfully...`);
```

Comment:
trivial: I think it should be "New datasource variable ${entryName}..." instead of ".. datasource property..."? if I am looking on context from naming at line of code 70?

Reply:
The suggestion to name it "Property" came from @rareddy .

Reply:
ok, just question for clarification. I was only wondering that at line 70 description is about value of variable and then at line 77 is success message about property, but still just trivial note. Thanks

Reply:
hah, good catch. Probably a copy / paste issue :) Will fix it.

Reply:
fixed that. Let me know if you find more.